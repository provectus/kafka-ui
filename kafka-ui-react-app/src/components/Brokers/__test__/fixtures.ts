import { BrokersLogdirs } from 'generated-sources';

export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com', port: 9092 },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com', port: 9092 },
];

export const clusterStatsPayload = {
  brokerCount: 2,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

const partition = {
  broker: 2,
  offsetLag: 0,
  partition: 2,
  size: 0,
};
const topics = {
  name: '_confluent-ksql-devquery_CTAS_NUMBER_OF_TESTS_59-Aggregate-Aggregate-Materialize-changelog',
  partitions: [partition],
};

export const brokerLogDirsPayload: BrokersLogdirs[] = [
  {
    error: 'NONE',
    name: '/opt/kafka/data-0/logs',
    topics: [
      {
        ...topics,
        partitions: [partition, partition, partition],
      },
      topics,
      {
        ...topics,
        partitions: [],
      },
    ],
  },
];
