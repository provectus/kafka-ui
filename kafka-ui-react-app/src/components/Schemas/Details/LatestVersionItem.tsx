import React from 'react';
import { SchemaSubject } from 'generated-sources';
import JSONEditor from 'components/common/JSONEditor/JSONEditor';

interface LatestVersionProps {
  schema: SchemaSubject;
}

const LatestVersionItem: React.FC<LatestVersionProps> = ({
  schema: { id, subject, schema, compatibilityLevel },
}) => (
  <div className="tile is-ancestor mt-1">
    <div className="tile is-4 is-parent">
      <div className="tile is-child">
        <table className="table is-fullwidth">
          <tbody>
            <tr>
              <td>ID</td>
              <td>{id}</td>
            </tr>
            <tr>
              <td>Subject</td>
              <td>{subject}</td>
            </tr>
            <tr>
              <td>Compatibility</td>
              <td>{compatibilityLevel}</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div className="tile is-parent">
      <div className="tile is-child box">
        <JSONEditor
          name="schema"
          value={JSON.stringify(JSON.parse(schema), null, '\t')}
          showGutter={false}
          readOnly
        />
      </div>
    </div>
  </div>
);

export default LatestVersionItem;
