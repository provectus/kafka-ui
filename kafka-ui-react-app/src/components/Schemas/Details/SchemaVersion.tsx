import React from 'react';
import { SchemaSubject } from 'generated-sources';
import JSONEditor from 'components/common/JSONEditor/JSONEditor';

interface SchemaVersionProps {
  version: SchemaSubject;
}

const SchemaVersion: React.FC<SchemaVersionProps> = ({
  version: { version, id, schema },
}) => {
  const [isOpen, setIsOpen] = React.useState(false);
  const toggleIsOpen = () => setIsOpen(!isOpen);
  return (
    <tr>
      <td>
        <span
          className="icon has-text-link is-size-7 is-small is-clickable"
          onClick={toggleIsOpen}
          aria-hidden
        >
          <i className={`fas fa-${isOpen ? 'minus' : 'plus'}`} />
        </span>
      </td>
      <td>{version}</td>
      <td>{id}</td>
      <td
        className="has-text-overflow-ellipsis is-family-code"
        style={{ width: '100%', maxWidth: 0 }}
      >
        {isOpen ? (
          <JSONEditor
            isFixedHeight
            name="schema"
            value={
              schema.trim().startsWith('{')
                ? JSON.stringify(JSON.parse(schema), null, '\t')
                : schema
            }
            setOptions={{
              showLineNumbers: false,
              maxLines: 40,
            }}
            readOnly
          />
        ) : (
          <span>{schema}</span>
        )}
      </td>
    </tr>
  );
};

export default SchemaVersion;
