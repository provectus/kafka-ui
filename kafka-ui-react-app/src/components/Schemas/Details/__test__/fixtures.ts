import { SchemaSubject, SchemaType } from 'generated-sources';
import {
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';

export const versionPayload = [
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
];
export const versionEmptyPayload = [];

export const jsonSchema: SchemaSubject = {
  subject: 'test',
  version: '15',
  id: 1,
  schema:
    '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.JSON,
};

export const protoSchema: SchemaSubject = {
  subject: 'test_proto',
  version: '1',
  id: 2,
  schema:
    'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.PROTOBUF,
};
