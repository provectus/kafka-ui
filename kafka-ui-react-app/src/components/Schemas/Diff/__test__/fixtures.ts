import { SchemaSubject, SchemaType } from 'generated-sources';

export const versions: SchemaSubject[] = [
  {
    subject: 'test',
    version: '3',
    id: 3,
    schema:
      'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.PROTOBUF,
  },
  {
    subject: 'test',
    version: '2',
    id: 2,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
];
