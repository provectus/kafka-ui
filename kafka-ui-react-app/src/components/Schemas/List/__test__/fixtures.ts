import { SchemaSubject, SchemaType } from 'generated-sources';

export const schemas: SchemaSubject[] = [
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
  {
    subject: 'test2',
    version: '1',
    id: 2,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
  {
    subject: 'test3',
    version: '1',
    id: 12,
    schema:
      '{"type":"record","name":"MyRecord3","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
];
