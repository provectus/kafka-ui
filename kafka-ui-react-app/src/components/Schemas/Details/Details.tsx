import React from 'react';
import { useNavigate } from 'react-router-dom';
import {
  ClusterSubjectParam,
  clusterSchemaEditPageRelativePath,
  clusterSchemaSchemaComparePageRelativePath,
  clusterSchemasPath,
} from 'lib/paths';
import ClusterContext from 'components/contexts/ClusterContext';
import PageLoader from 'components/common/PageLoader/PageLoader';
import PageHeading from 'components/common/PageHeading/PageHeading';
import { Button } from 'components/common/Button/Button';
import { useAppDispatch, useAppSelector } from 'lib/hooks/redux';
import {
  fetchLatestSchema,
  fetchSchemaVersions,
  getAreSchemaLatestFulfilled,
  getAreSchemaVersionsFulfilled,
  SCHEMAS_VERSIONS_FETCH_ACTION,
  SCHEMA_LATEST_FETCH_ACTION,
  selectAllSchemaVersions,
  getSchemaLatest,
  getAreSchemaLatestRejected,
} from 'redux/reducers/schemas/schemasSlice';
import { showServerError } from 'lib/errorHandling';
import { resetLoaderById } from 'redux/reducers/loader/loaderSlice';
import { TableTitle } from 'components/common/table/TableTitle/TableTitle.styled';
import useAppParams from 'lib/hooks/useAppParams';
import { schemasApiClient } from 'lib/api';
import { Dropdown } from 'components/common/Dropdown';
import Table from 'components/common/NewTable';
import { Action, ResourceType } from 'generated-sources';
import {
  ActionButton,
  ActionDropdownItem,
} from 'components/common/ActionComponent';

import LatestVersionItem from './LatestVersion/LatestVersionItem';
import SchemaVersion from './SchemaVersion/SchemaVersion';

const Details: React.FC = () => {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const { isReadOnly } = React.useContext(ClusterContext);
  const { clusterName, subject } = useAppParams<ClusterSubjectParam>();

  React.useEffect(() => {
    dispatch(fetchLatestSchema({ clusterName, subject }));
    return () => {
      dispatch(resetLoaderById(SCHEMA_LATEST_FETCH_ACTION));
    };
  }, [clusterName, dispatch, subject]);

  React.useEffect(() => {
    dispatch(fetchSchemaVersions({ clusterName, subject }));
    return () => {
      dispatch(resetLoaderById(SCHEMAS_VERSIONS_FETCH_ACTION));
    };
  }, [clusterName, dispatch, subject]);

  const versions = useAppSelector((state) => selectAllSchemaVersions(state));
  const schema = useAppSelector(getSchemaLatest);
  const isFetched = useAppSelector(getAreSchemaLatestFulfilled);
  const isRejected = useAppSelector(getAreSchemaLatestRejected);
  const areVersionsFetched = useAppSelector(getAreSchemaVersionsFulfilled);

  const columns = React.useMemo(
    () => [
      { header: 'Version', accessorKey: 'version' },
      { header: 'ID', accessorKey: 'id' },
      { header: 'Type', accessorKey: 'schemaType' },
    ],
    []
  );

  const deleteHandler = async () => {
    try {
      await schemasApiClient.deleteSchema({
        clusterName,
        subject,
      });
      navigate('../');
    } catch (e) {
      showServerError(e as Response);
    }
  };

  if (isRejected) {
    navigate('/404');
  }

  if (!isFetched || !schema) {
    return <PageLoader />;
  }
  return (
    <>
      <PageHeading
        text={schema.subject}
        backText="Schema Registry"
        backTo={clusterSchemasPath(clusterName)}
      >
        {!isReadOnly && (
          <>
            <Button
              buttonSize="M"
              buttonType="primary"
              to={{
                pathname: clusterSchemaSchemaComparePageRelativePath,
                search: `leftVersion=${versions[0]?.version}&rightVersion=${versions[0]?.version}`,
              }}
            >
              Compare Versions
            </Button>
            <ActionButton
              buttonSize="M"
              buttonType="primary"
              to={clusterSchemaEditPageRelativePath}
              permission={{
                resource: ResourceType.SCHEMA,
                action: Action.EDIT,
                value: subject,
              }}
            >
              Edit Schema
            </ActionButton>
            <Dropdown>
              <ActionDropdownItem
                confirm={
                  <>
                    Are you sure want to remove <b>{subject}</b> schema?
                  </>
                }
                onClick={deleteHandler}
                danger
                permission={{
                  resource: ResourceType.SCHEMA,
                  action: Action.DELETE,
                  value: subject,
                }}
              >
                Remove schema
              </ActionDropdownItem>
            </Dropdown>
          </>
        )}
      </PageHeading>
      <LatestVersionItem schema={schema} />
      <TableTitle>Old versions</TableTitle>
      {areVersionsFetched ? (
        <Table
          columns={columns}
          data={versions}
          getRowCanExpand={() => true}
          renderSubComponent={SchemaVersion}
          enableSorting
        />
      ) : (
        <PageLoader />
      )}
    </>
  );
};

export default Details;
