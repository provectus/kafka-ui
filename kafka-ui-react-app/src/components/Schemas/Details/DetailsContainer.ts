import { connect } from 'react-redux';
import { ClusterName, RootState } from 'redux/interfaces';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import {
  getIsSchemaVersionFetched,
  getSchema,
  getSortedSchemaVersions,
} from 'redux/reducers/schemas/selectors';
import { fetchSchemaVersions } from 'redux/actions';
import Details from './Details';

interface RouteProps {
  clusterName: ClusterName;
  subject: string;
}

type OwnProps = RouteComponentProps<RouteProps>;

const mapStateToProps = (
  state: RootState,
  {
    match: {
      params: { clusterName, subject },
    },
  }: OwnProps
) => ({
  subject,
  schema: getSchema(state, subject),
  versions: getSortedSchemaVersions(state),
  isFetched: getIsSchemaVersionFetched(state),
  clusterName,
});

const mapDispatchToProps = {
  fetchSchemaVersions,
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Details)
);
