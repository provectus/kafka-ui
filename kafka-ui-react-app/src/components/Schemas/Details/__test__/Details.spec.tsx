import React from 'react';
import Details from 'components/Schemas/Details/Details';
import { render, WithRoute } from 'lib/testHelpers';
import { clusterSchemaPath } from 'lib/paths';
import { screen } from '@testing-library/dom';
import {
  schemasInitialState,
  schemaVersion,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';
import fetchMock from 'fetch-mock';
import ClusterContext, {
  ContextProps,
  initialValue as contextInitialValue,
} from 'components/contexts/ClusterContext';
import { RootState } from 'redux/interfaces';
import { act } from '@testing-library/react';

import { versionPayload, versionEmptyPayload } from './fixtures';

const clusterName = 'testClusterName';
const schemasAPILatestUrl = `/api/clusters/${clusterName}/schemas/${schemaVersion.subject}/latest`;
const schemasAPIVersionsUrl = `/api/clusters/${clusterName}/schemas/${schemaVersion.subject}/versions`;

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockHistoryPush,
}));

const renderComponent = (
  initialState: RootState['schemas'] = schemasInitialState,
  context: ContextProps = contextInitialValue
) =>
  render(
    <WithRoute path={clusterSchemaPath()}>
      <ClusterContext.Provider value={context}>
        <Details />
      </ClusterContext.Provider>
    </WithRoute>,
    {
      initialEntries: [clusterSchemaPath(clusterName, schemaVersion.subject)],
      preloadedState: {
        schemas: initialState,
      },
    }
  );

describe('Details', () => {
  afterEach(() => fetchMock.reset());

  describe('fetch failed', () => {
    it('renders pageloader', async () => {
      const schemasAPILatestMock = fetchMock.getOnce(schemasAPILatestUrl, 404);
      const schemasAPIVersionsMock = fetchMock.getOnce(
        schemasAPIVersionsUrl,
        404
      );
      await act(() => {
        renderComponent();
      });
      expect(schemasAPILatestMock.called(schemasAPILatestUrl)).toBeTruthy();
      expect(schemasAPIVersionsMock.called(schemasAPIVersionsUrl)).toBeTruthy();
      expect(screen.getByRole('progressbar')).toBeInTheDocument();
      expect(screen.queryByText(schemaVersion.subject)).not.toBeInTheDocument();
      expect(screen.queryByText('Edit Schema')).not.toBeInTheDocument();
      expect(screen.queryByText('Remove Schema')).not.toBeInTheDocument();
    });
  });

  describe('fetch success', () => {
    describe('has schema versions', () => {
      it('renders component with schema info', async () => {
        const schemasAPILatestMock = fetchMock.getOnce(
          schemasAPILatestUrl,
          schemaVersion
        );
        const schemasAPIVersionsMock = fetchMock.getOnce(
          schemasAPIVersionsUrl,
          versionPayload
        );
        await act(() => {
          renderComponent();
        });
        expect(schemasAPILatestMock.called()).toBeTruthy();
        expect(schemasAPIVersionsMock.called()).toBeTruthy();
        expect(screen.getByText('Edit Schema')).toBeInTheDocument();
        expect(screen.queryByRole('progressbar')).not.toBeInTheDocument();
        expect(screen.getByRole('table')).toBeInTheDocument();
      });
    });

    describe('fetch success schema with non ascii characters', () => {
      describe('has schema versions', () => {
        it('renders component with schema info', async () => {
          const schemasAPILatestMock = fetchMock.getOnce(
            schemasAPILatestUrl,
            schemaVersionWithNonAsciiChars
          );
          const schemasAPIVersionsMock = fetchMock.getOnce(
            schemasAPIVersionsUrl,
            versionPayload
          );
          await act(() => {
            renderComponent();
          });
          expect(schemasAPILatestMock.called()).toBeTruthy();
          expect(schemasAPIVersionsMock.called()).toBeTruthy();
          expect(screen.getByText('Edit Schema')).toBeInTheDocument();
          expect(screen.queryByRole('progressbar')).not.toBeInTheDocument();
          expect(screen.getByRole('table')).toBeInTheDocument();
        });
      });
    });

    describe('empty schema versions', () => {
      beforeEach(async () => {
        const schemasAPILatestMock = fetchMock.getOnce(
          schemasAPILatestUrl,
          schemaVersion
        );
        const schemasAPIVersionsMock = fetchMock.getOnce(
          schemasAPIVersionsUrl,
          versionEmptyPayload
        );
        await act(() => {
          renderComponent();
        });
        expect(schemasAPILatestMock.called()).toBeTruthy();
        expect(schemasAPIVersionsMock.called()).toBeTruthy();
      });

      // seems like incorrect behaviour
      it('renders versions table with 0 items', () => {
        expect(screen.getByRole('table')).toBeInTheDocument();
      });
    });
  });
});
