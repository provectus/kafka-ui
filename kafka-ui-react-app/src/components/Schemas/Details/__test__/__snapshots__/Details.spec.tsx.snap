// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details View Initial state matches snapshot 1`] = `
<div
  className="section"
>
  <div
    className="level"
  >
    <Breadcrumb
      links={
        Array [
          Object {
            "href": "/ui/clusters/Test cluster/schemas",
            "label": "Schema Registry",
          },
        ]
      }
    >
      test
    </Breadcrumb>
  </div>
  <div
    className="box"
  >
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <div
            className="mr-1"
          >
            <b>
              Latest Version
            </b>
          </div>
          <div
            className="tag is-info is-light"
            title="Version"
          >
            #
            1
          </div>
        </div>
      </div>
      <div
        className="level-right"
      >
        <button
          className="button is-warning is-small level-item"
          disabled={true}
          title="in development"
          type="button"
        >
          Update Schema
        </button>
        <button
          className="button is-danger is-small level-item"
          disabled={true}
          title="in development"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
    <LatestVersionItem
      schema={
        Object {
          "compatibilityLevel": "BACKWARD",
          "id": 1,
          "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord1\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
          "schemaType": "JSON",
          "subject": "test",
          "version": "1",
        }
      }
    />
  </div>
  <div
    className="box"
  >
    <table
      className="table is-striped is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            Version
          </th>
          <th>
            ID
          </th>
          <th>
            Schema
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Details View when page with schema versions is loading matches snapshot 1`] = `
<div
  className="section"
>
  <div
    className="level"
  >
    <Breadcrumb
      links={
        Array [
          Object {
            "href": "/ui/clusters/Test cluster/schemas",
            "label": "Schema Registry",
          },
        ]
      }
    >
      test
    </Breadcrumb>
  </div>
  <PageLoader />
</div>
`;

exports[`Details View when page with schema versions loaded when schema has versions matches snapshot 1`] = `
<div
  className="section"
>
  <div
    className="level"
  >
    <Breadcrumb
      links={
        Array [
          Object {
            "href": "/ui/clusters/Test cluster/schemas",
            "label": "Schema Registry",
          },
        ]
      }
    >
      test
    </Breadcrumb>
  </div>
  <div
    className="box"
  >
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <div
            className="mr-1"
          >
            <b>
              Latest Version
            </b>
          </div>
          <div
            className="tag is-info is-light"
            title="Version"
          >
            #
            1
          </div>
        </div>
      </div>
      <div
        className="level-right"
      >
        <button
          className="button is-warning is-small level-item"
          disabled={true}
          title="in development"
          type="button"
        >
          Update Schema
        </button>
        <button
          className="button is-danger is-small level-item"
          disabled={true}
          title="in development"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
    <LatestVersionItem
      schema={
        Object {
          "compatibilityLevel": "BACKWARD",
          "id": 1,
          "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord1\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
          "schemaType": "JSON",
          "subject": "test",
          "version": "1",
        }
      }
    />
  </div>
  <div
    className="box"
  >
    <table
      className="table is-striped is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            Version
          </th>
          <th>
            ID
          </th>
          <th>
            Schema
          </th>
        </tr>
      </thead>
      <tbody>
        <SchemaVersion
          key="1"
          version={
            Object {
              "compatibilityLevel": "BACKWARD",
              "id": 1,
              "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord1\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
              "schemaType": "JSON",
              "subject": "test",
              "version": "1",
            }
          }
        />
        <SchemaVersion
          key="2"
          version={
            Object {
              "compatibilityLevel": "BACKWARD",
              "id": 2,
              "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord2\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
              "schemaType": "JSON",
              "subject": "test",
              "version": "2",
            }
          }
        />
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Details View when page with schema versions loaded when versions are empty matches snapshot 1`] = `
<div
  className="section"
>
  <div
    className="level"
  >
    <Breadcrumb
      links={
        Array [
          Object {
            "href": "/ui/clusters/Test cluster/schemas",
            "label": "Schema Registry",
          },
        ]
      }
    >
      test
    </Breadcrumb>
  </div>
  <div
    className="box"
  >
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <div
            className="mr-1"
          >
            <b>
              Latest Version
            </b>
          </div>
          <div
            className="tag is-info is-light"
            title="Version"
          >
            #
            1
          </div>
        </div>
      </div>
      <div
        className="level-right"
      >
        <button
          className="button is-warning is-small level-item"
          disabled={true}
          title="in development"
          type="button"
        >
          Update Schema
        </button>
        <button
          className="button is-danger is-small level-item"
          disabled={true}
          title="in development"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
    <LatestVersionItem
      schema={
        Object {
          "compatibilityLevel": "BACKWARD",
          "id": 1,
          "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord1\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
          "schemaType": "JSON",
          "subject": "test",
          "version": "1",
        }
      }
    />
  </div>
  <div
    className="box"
  >
    <table
      className="table is-striped is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            Version
          </th>
          <th>
            ID
          </th>
          <th>
            Schema
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
